lib/web_ui/lib/src/engine/skwasm/skwasm_impl/memory.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:js_interop';
+
+import 'package:ui/src/engine.dart';
+
+abstract class SkwasmObjectWrapper<T extends NativeType> {
+  Pointer<T> get handle;
+}
+
+typedef DisposeFunction<T extends NativeType> = void Function(Pointer<T>);
+
+class SkwasmFinalizationRegistry<T extends NativeType> {
```

Comment:
Do we actually need the `<T>` type parameter? If not, then I think we can use a shared `DomFinalizationRegistry` instance for all objects. But, again, I'm not sure what kind of tradeoff exists in using one vs many registries 🤷  Just pointing out a possibility.

Reply:
We do need this type parameter, because each finalization registry has its own disposal function that it invokes and that disposal function takes a specific pointer type. I mentioned this in the other comment that it is possible to do this with another wrapper which basically "type-erases" this problem away, but I'd rather not add another layer of indirection.